controllers/chat/moveToAnon.js

```diff
@@ -0,0 +1,130 @@
+const {User, ChatAnonUserInfo} = require('../../databases/models');
+const {StreamChat} = require('stream-chat');
+const ErrorResponse = require('../../utils/response/ErrorResponse');
+const {responseError, responseSuccess} = require('../../utils/Responses');
+const {CHANNEL_TYPE} = require('../../helpers/constants');
+const UsersFunction = require('../../databases/functions/users');
+
+/**
+ *
+ * @param {import("express").Request} req
+ * @param {import("express").Response} res
+ */
+const moveToAnon = async (req, res) => {
+  const {
+    targetUser,
+    oldChannelId,
+    anon_user_info_color_code,
+    anon_user_info_color_name,
+    anon_user_info_emoji_code,
+    anon_user_info_emoji_name
+  } = req.body;
+
+  let members = [targetUser];
+  if (!members.includes(req.userId)) members.push(req.userId);
+
+  const client = StreamChat.getInstance(process.env.API_KEY, process.env.SECRET);
+  try {
+    /**
+     * @type {import('stream-chat').OwnUserResponse}
+     */
+    const user = {
+      name: `Anonymous ${anon_user_info_emoji_name}`,
+      id: req.userId,
+      image: '',
+      username: `Anonymous ${anon_user_info_emoji_name}`
+    };
+    await client.connectUser(user, req.token);
+
+    if (client.user.name !== `Anonymous ${anon_user_info_emoji_name}`) {
+      await client.upsertUser({id: req.userId, name: `Anonymous ${anon_user_info_emoji_name}`});
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There's a potential issue here where you're connecting the user and then immediately checking if the user's name matches the anonymous name. However, there's no guarantee that `client.user.name` will be updated immediately after `client.connectUser()` is called because it's an asynchronous operation. This could lead to unnecessary calls to `client.upsertUser()`. 



<!-- This is an auto-generated comment by OSS CodeRabbit -->